src/default_client.rs

```diff
@@ -742,60 +1256,162 @@ impl<A: ClientAdapter + Sync + Send> Client for DefaultClient<A> {
 
         self.send(request).await
     }
 
+    /// List all users part of the group.
+    ///
+    /// It is possible to filter users based on their state. See [Groups and Clans](https://heroiclabs.com/docs/social-groups-clans/#groups-and-clans)
+    /// for possible values.
+    ///
+    /// See [Limit and cursor](index.html#limit-and-cursor) for a description on how to use the `limit` and `cursor` parameters.
+    ///
+    /// # Example
+    /// ```
+    /// # #![feature(async_closure)]
+    /// # use nakama_rs::test_helpers::*;
+    /// # run_in_example(async move |client, session| {
+    /// let group = client.create_group(&session, "NewGroup", None, None, None, None, None).await?;
+    /// client.list_group_users(&session, &group.id, None, None, None).await
+    ///     .expect("Failed to list group users");
+    /// # Ok(())
+    /// # })
+    /// ```    
     async fn list_group_users(
         &self,
-        session: &mut Session,
+        session: &Session,
         group_id: &str,
         state: Option<i32>,
         limit: Option<i32>,
         cursor: Option<&str>,
     ) -> Result<ApiGroupUserList, Self::Error> {
-        let request = api::list_group_users(&session.auth_token, group_id, limit, state, cursor);
+        let request =
+            api::list_group_users(&session.get_auth_token(), group_id, limit, state, cursor);
 
         self.send(request).await
     }
 
+    /// List groups on the server.
+    ///
+    /// It is possible to filter groups by name. The percentage char '%' can be used as placeholder
+    ///
+    /// See [Limit and cursor](index.html#limit-and-cursor) for a description on how to use the `limit` and `cursor` parameters.
+    ///
+    /// # Example
+    /// ```
+    /// # #![feature(async_closure)]
+    /// # use nakama_rs::test_helpers::*;
+    /// # run_in_example(async move |client, session| {
+    /// // List all groups
+    /// client.list_groups(&session, None, None, None).await
+    ///     .expect("Failed to list groups");
+    /// // Search for groups starting with "Instance1"
+    /// client.list_groups(&session, Some("Instance1%"), None, None).await
+    ///     .expect("Failed to list groups");
+    /// # Ok(())
+    /// # })
+    /// ```    
     async fn list_groups(
         &self,
-        session: &mut Session,
+        session: &Session,
         name: Option<&str>,
         limit: Option<i32>,
         cursor: Option<&str>,
     ) -> Result<ApiGroupList, Self::Error> {
-        let request = api::list_groups(&session.auth_token, name, cursor, limit);
+        let request = api::list_groups(&session.get_auth_token(), name, cursor, limit);
 
         self.send(request).await
     }
 
+    /// List records from a leaderboard
+    ///
+    /// If you specify `owner_ids` only records of those users will be returned.
+    ///
+    /// Set `expiry` in seconds (since epoch) to fetch older records. Optional. 0 means active leaderboards.
+    /// TODO: Verify this is correct.
```

Comment:
I'm not sure how `expiry` is supposed to wrok.

Reply:
Leaderboards can be scheduled to recur. So if you want to grab the records for specific instances of that leaderboard in the past, then you can specify the `expiry`.

Reply:
Is it okay if I copy your explanation 1:1?